place_tracker/lib/place_tracker_app.dart

```diff
@@ -57,80 +50,58 @@ class _PlaceTrackerHomePage extends StatelessWidget {
             padding: EdgeInsets.fromLTRB(0.0, 0.0, 16.0, 0.0),
             child: IconButton(
               icon: Icon(
-                AppState.of(context).viewType == PlaceTrackerViewType.map
+                state.viewType == PlaceTrackerViewType.map
                     ? Icons.list
                     : Icons.map,
                 size: 32.0,
               ),
               onPressed: () {
-                AppState.updateWith(
-                  context,
-                  viewType:
-                      AppState.of(context).viewType == PlaceTrackerViewType.map
-                          ? PlaceTrackerViewType.list
-                          : PlaceTrackerViewType.map,
+                state.changeViewType(
+                  state.viewType == PlaceTrackerViewType.map
+                      ? PlaceTrackerViewType.list
+                      : PlaceTrackerViewType.map,
                 );
               },
             ),
           ),
         ],
       ),
       body: IndexedStack(
-        index:
-            AppState.of(context).viewType == PlaceTrackerViewType.map ? 0 : 1,
+        index: state.viewType == PlaceTrackerViewType.map ? 0 : 1,
         children: <Widget>[
           PlaceMap(center: const LatLng(45.521563, -122.677433)),
-          PlaceList(),
+          PlaceList()
         ],
       ),
     );
   }
 }
 
-class AppState {
-  const AppState({
+class AppState with ChangeNotifier {
```

Comment:
It's purely a convention, but we typically use `extends` rather than `with` to integrate ChangeNotifier.